gap/attributes/homomorph.gi

```diff
@@ -0,0 +1,495 @@
+#############################################################################
+##
+##  homomorph.gi
+##  Copyright (C) 2022                               Artemis Konstantinidi
+##                                                         Chinmaya Nagpal
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+# This file contains various methods for representing homomorphisms between
+# semigroups
+
+InstallMethod(SemigroupHomomorphismByImages, "for two semigroups and two lists",
+[IsSemigroup, IsSemigroup, IsList, IsList],
+function(S, T, gens, imgs)
+  local map, R, rel, original_gens, U;
+
+  if not ForAll(gens, x -> x in S) then
+    ErrorNoReturn("the 3rd argument (a list) must consist of elements ",
+                  "of the 1st argument (a semigroup)");
+  elif Semigroup(gens) <> S then
+    ErrorNoReturn("the 1st argument (a semigroup) is not generated by ",
+                  "the 3rd argument (a list)");
+  elif not ForAll(imgs, x -> x in T) then
+    ErrorNoReturn("the 4th argument (a list) must consist of elements ",
+                  "of the 2nd argument (a semigroup)");
+  elif Size(gens) <> Size(imgs) then
+    ErrorNoReturn("the 3rd argument (a list) and the 4th argument ",
+                  "(a list) are not the same size");
+  fi;
+
+  # in case of different generators, do:
+  # gens -> original generators (as passed to Semigroup function)
+  # imgs -> images of original generators
+  U := Semigroup(gens, rec(acting := true));
```

Comment:
I'm not sure why this line is here at all, and in particular, why the `acting := true` is here. For many types of semigroups this won't do anything at all, so it can't be necessary.

Reply:
Is this added because you need `Factorization` to be implemented? If so, then the filter for the method ought to be `IsSemigroup and CanUseFroduirePin` rather than just `IsSemigroup`.